src/viam/sdk/spatialmath/geometry.cpp

```diff
@@ -94,7 +111,62 @@ viam::common::v1::Geometry GeometryConfig::to_proto() const {
             return geometry_;
         }
     }
-};
+}
+void GeometryConfig::set_coordinates(GeometryConfig::coordinates coord) {
+    x_ = coord.x;
+    y_ = coord.y;
+    z_ = coord.z;
+}
+
+void GeometryConfig::set_pose(GeometryConfig::pose pos) {
+    o_x_ = pos.x;
+    o_y_ = pos.y;
+    o_z_ = pos.z;
+}
+void GeometryConfig::set_radius(double r) {
+    r_ = r;
+}
+void GeometryConfig::set_geometry_type(GeometryType type) {
+    geometry_type_ = type;
+}
+void GeometryConfig::set_orientation_config(OrientationConfig config) {
+    orientation_config_ = config;
+}
+void GeometryConfig::set_label(std::string label) {
+    label_ = label;
+}
+GeometryConfig::coordinates GeometryConfig::get_coordinates() const {
+    return {x_, y_, z_};
+}
+
+GeometryConfig::pose GeometryConfig::get_pose() const {
+    return {o_x_, o_y_, o_z_};
+}
+
+double GeometryConfig::get_radius() const {
+    return r_;
+}
+
+GeometryType GeometryConfig::get_geometry_type() const {
+    return geometry_type_;
+}
+
+OrientationConfig GeometryConfig::get_orientation_config() const {
+    return orientation_config_;
+}
+
+std::string GeometryConfig::get_label() const {
+    return label_;
+}
+
+bool operator==(const GeometryConfig& lhs, const GeometryConfig& rhs) {
+    if (lhs.geometry_type_ == GeometryType::sphere && lhs.r_ != rhs.r_) {
```

Comment:
might be good to add a quick comment explaining the reason for this particular bit of logic.